### PR TITLE
Geckos Story 1

### PR DESCRIPTION
**Pull request by jdselby24**
Added functionality to link a hiring partner to an event

### CODE REVIEW

File: public/js/eventsPage.js [hunk 1]
```diff
@@ -34,8 +36,71 @@ function displayEventsHandler(events) {
...
+                let showInfoButtons = document.querySelectorAll('.show-event-info')
+            showInfoButtons.forEach(function (button) {
+                button.addEventListener('click', e => {
+                    let targetId = 'moreInfo' + e.target.dataset.reference
+                    let targetDiv = document.getElementById(targetId)
+                    targetDiv.classList.toggle('hide')     
+                    })
+                })
+            }).then(()=>{
+                let HPForms = document.querySelectorAll('.addHiringPartnerForm')
```
Comment (charliecog) [inline public/js/eventsPage.js]: variable name should be camelCase

File: public/js/eventsPage.js [hunk 2]
```diff
@@ -34,8 +36,71 @@ function displayEventsHandler(events) {
...
+                            attendees = 0
+                        }
+
+                        let data = {
+                            hiring_partner_id: hpId,
+                            events_id: eventIdForm,
+                            people_attending: attendees
+                        }
+
+                        if(!(attendees < 0)) {
```
Comment (charliecog) [inline public/js/eventsPage.js]: Double negative just use > 0

File: public/js/eventsPage.js [hunk 3]
```diff
@@ -34,8 +36,71 @@ function displayEventsHandler(events) {
...
+                })
+            })
+        })   
+    }
+};
+
+/**
+ * Outputs HTML elements with event details
+ *
+ * @param evens an object which contains information about an event
```
Comment (charliecog) [inline public/js/eventsPage.js]: spello evens

File: src/Classes/Controllers/LinkHiringPartnerToEventController.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Portal\Controllers;
+
+use Portal\Models\EventModel;
+use Slim\Http\Request;
+use Slim\Http\Response;
+
+class LinkHiringPartnerToEventController
```
Comment (charliecog) [inline src/Classes/Controllers/LinkHiringPartnerToEventController.php]: AddHiringPartnerToEventController makes more sense as a name

File: src/Classes/Controllers/LinkHiringPartnerToEventController.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        $data = $request->getParsedBody();
+        $hiringPartner = $data['hiring_partner_id'];
+        $event = $data['events_id'];
+        $attendees = $data['people_attending'];
+        $result = $this->eventModel->linkHPToEvent($hiringPartner, $event, $attendees);
+        if ($result) {
+            return $response->withJson(['success' => true,
+                'message'=>'Hiring partner successfully linked to event.'], 200);
+        } else {
+            return $response->withJson(['success' => false, 'message'=>'Database error!'], 500);
```
Comment (charliecog) [inline src/Classes/Controllers/LinkHiringPartnerToEventController.php]: Just say error - dont tell the user about the Database - they dont care/too much info to give out

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 6]
```diff
@@ -2,8 +2,11 @@
 
 namespace Portal\Entities;
 
+use phpDocumentor\Reflection\Types\Integer;
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: que?

File: src/Classes/Models/EventModel.php [hunk 7]
```diff
@@ -2,6 +2,8 @@
 
 namespace Portal\Models;
 
+use phpDocumentor\Reflection\Types\Boolean;
```
Comment (charliecog) [inline src/Classes/Models/EventModel.php]: huh?

File: src/Classes/Models/EventModel.php [hunk 8]
```diff
@@ -2,6 +2,8 @@
...
+use phpDocumentor\Reflection\Types\Integer;
```
Comment (charliecog) [inline src/Classes/Models/EventModel.php]: what?

File: src/routes.php [hunk 9]
```diff
@@ -21,6 +21,8 @@
 $app->get('/api/getEvents', 'GetEventsController');
 $app->post('/api/addEvent', 'AddEventController');
 $app->post('/api/addContact', 'AddContactController');
+$app->post('/api/linkHiringPartnerToEvent', 'LinkHiringPartnerToEventController');
```
Comment (charliecog) [inline src/routes.php]: again naming - addHiringPartnerToEvent makes more sense

File: public/js/eventsPage.js (hunk 1 above)
Comment (sudokufan) [reply public/js/eventsPage.js]: fixed

File: public/js/eventsPage.js (hunk 3 above)
Comment (sudokufan) [reply public/js/eventsPage.js]: done

File: src/Classes/Entities/HiringPartnerEntity.php (hunk 6 above)
Comment (sudokufan) [reply src/Classes/Entities/HiringPartnerEntity.php]: removed

File: src/Classes/Models/EventModel.php (hunk 7 above)
Comment (sudokufan) [reply src/Classes/Models/EventModel.php]: removed

File: src/Classes/Models/EventModel.php (hunk 8 above)
Comment (sudokufan) [reply src/Classes/Models/EventModel.php]: who knows mate. Removed

File: public/js/eventsPage.js (hunk 2 above)
Comment (mgallagher56) [reply public/js/eventsPage.js]: Fixed

File: src/Classes/Controllers/LinkHiringPartnerToEventController.php (hunk 5 above)
Comment (sudokufan) [reply src/Classes/Controllers/LinkHiringPartnerToEventController.php]: done

File: src/Classes/Controllers/AddHiringPartnerToEventController.php [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    public function __invoke(Request $request, Response $response, array $args): Response
+    {
+        $data = $request->getParsedBody();
+        $hiringPartner = $data['hiring_partner_id'];
+        $event = $data['event_id'];
+        $attendees = $data['people_attending'];
+
+        if (!$this->eventModel->checklinkHP($hiringPartner, $event)) {
+            $result = $this->eventModel->linkHPToEvent($hiringPartner, $event, $attendees);
```
Comment (charliecog) [inline src/Classes/Controllers/AddHiringPartnerToEventController.php]: this still says link HP to Event?

Comment (sudokufan) [reply src/Classes/Controllers/AddHiringPartnerToEventController.php]: nah mate

**Pull request merged**